### ISSUE TITLE
UiIconButton does not play well with KIcon

### ISSUE THREAD

**Issue by nucleogenesis** [learningequality/kolibri#5713]
### Observed behavior

When using a `<KIcon/>` inside of a `<UiIconButton />` component, the `<KIcon />` does not display as expected. 

Example:

```html
<UiIconButton type="primary" size="small">
  <KIcon back />
</UiIconButton>
```
In this instance, the arrow icon is not vertically centered. The `<KIcon />` requires some CSS to adjust it properly.

`<KIcon back style="position: relative; top: -1px;" />` - in this case fixes the display.

### Expected behavior

Icons ought to display properly without additional CSS styles passed to the `<KIcon />` inside of a `<UiIconButton></UiIconButton>` component.

### User-facing consequences

Only a user-facing issue if a developer fails to notice and accommodate for the styling issues.

### Steps to reproduce

This branch includes the CSS that fixes the display, you can check this out and remove that CSS to see the issue in action: https://github.com/nucleogenesis/kolibri/tree/facility-user-search-speed

**Comment by indirectlylit** [comment]
`UiIconButton` is a hold-over from Keen. We should fork it and re-implement it as `KIconButton`

In order to reduce boilerplate for the most common use case, we can add a prop which takes an icon name as an alternative passing in the icon through a slot. This would automatically instantiate a `KIcon` internally.

For example,

```html
<KIconButton icon="back" />
```

would be equivalent to 

```html
<KIconButton>
  <KIcon back />
</KIconButton>
```

See also #5714 regarding changing the `KIcon` API to `<KIcon icon="back" />`

**Comment by nucleogenesis** [comment]
This has been fixed across multiple PRs over some time between the Kolibri Design System and Kolibri and is set to be released with Kolibri 0.14 and the first KDS release in tandem.
